tests/legacy-cli/e2e/tests/build/auto-csp.ts

```diff
@@ -0,0 +1,131 @@
+import assert from 'node:assert';
+import { getGlobalVariable } from '../../utils/env';
+import { expectFileToMatch, rimraf, writeMultipleFiles } from '../../utils/fs';
+import { findFreePort } from '../../utils/network';
+import { installWorkspacePackages } from '../../utils/packages';
+import { execAndWaitForOutputToMatch, ng } from '../../utils/process';
+import { updateJsonFile, updateServerFileForWebpack, useSha } from '../../utils/project';
+
+const MULTI_HASH_CSP =
+  /script-src 'strict-dynamic' (?:'sha256-[^']+' )+https: 'unsafe-inline';object-src 'none';base-uri 'self';/;
+
+export default async function () {
+  assert(
+    getGlobalVariable('argv')['esbuild'],
+    'This test should not be called in the Webpack suite.',
+  );
+  // forcibly remove in case another test doesn't clean itself up
+  // await rimraf('node_modules/@angular/ssr');
+  // We don't need SSR yet but eventually for when Auto-CSP will also have a SSR implementation.
+  // await ng('add', '@angular/ssr', '--skip-confirmation', '--skip-install');
+
+  await useSha();
+  await installWorkspacePackages();
```

Comment:
This can also be commented out for now since it is only need for the above `@angular/ssr` installation.